packages/base-dao/src/baseDAO.ts

```diff
@@ -402,8 +409,12 @@ function logRequestsThatReturnUnauthorizedEntities(entitiesThatUserCannotAccess,
   }
 }
 
+function allDataEqual(dataArray) {
+  return dataArray.every(data => isEqual(dataArray[0], data));
```

Comment:
Don't think this will work. The entities here were fetched from the DB. They will never be fully equal.

Reply:
Need to add tests.

This use case as well with veryfing that it calls the update many when the update is the same for many entities. My assumption is that it will fail here.

Reply:
One alternative is to see if the presave changed anything. For each entity compare pre and post save entities and save a signal boolean saying if any of them changed. If any of them did, assume you need to go one by one, if not, update many

Reply:
What's returned from here is not the entity that fetched from the db , it's the data that needs to be updated.Like if I pass status to some entity and then change the status to something else in preSave.I've added test and it's working.